Yosemite/Yosemite/Stores/Helpers/SiteSpecificAppSettingsStoreMethods.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+import Storage
+
+protocol SiteSpecificAppSettingsStoreMethodsProtocol {
+    func getStoreSettings(for siteID: Int64) -> GeneralStoreSettings
+    func setStoreSettings(settings: GeneralStoreSettings, for siteID: Int64, onCompletion: ((Result<Void, Error>) -> Void)?)
+    func resetStoreSettings()
+    func setStoreID(siteID: Int64, id: String?)
+    func getStoreID(siteID: Int64, onCompletion: (String?) -> Void)
+}
+
+/// Methods for managing site-specific app settings
+///
+public struct SiteSpecificAppSettingsStoreMethods: SiteSpecificAppSettingsStoreMethodsProtocol {
```

Comment:
> This PR extracts the site-specific local app settings storage interface, so it can be used elsewhere in Yosemite apart from the AppSettingsStorage.

We could keep it `internal` if we only intend to use it within `Yosemite`. At least that was the choice I made when moving `coupon` and `settingStore` methods.